### PR TITLE
Harden the OpenVPN server AMI

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request makes several changes related to hardening the OpenVPN server AMI according to the CIS guidance.

## 💭 Motivation and Context

We are required to harden the OpenVPN and FreeIPA AMIs used in the COOL to obtain our ATO.

## 🧪 Testing

These changes have been used to successfully deploy a hardened OpenVPN EC2 instance in our staging COOL environment.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/harden.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+- hosts: all
+  name: Hardening tasks
+  become: yes
+  become_method: sudo
+  roles:
+    - role: harden
+      # vars:
+      #   sshd_admin_net:
+      #     - 127.0.0.1/32
```
Comment (dav3r) [inline src/harden.yml:10]: @jsf9k - Do we want to remove these commented-out lines?

**Review by dav3r — APPROVED** [review]
HARDENED! 🏗️ 

Please note my one question.

**Review by hillaryj — APPROVED** [review]
🚑

File: src/requirements.yml [hunk 2]
```diff
@@ -10,12 +12,15 @@
...
   name: freeipa_client
+- src: https://github.com/cisagov/ansible-role-hardening
+  name: harden
 - src: https://github.com/cisagov/ansible-role-htop
   name: htop
 - src: https://github.com/cisagov/ansible-role-nvme
   name: nvme
 - src: https://github.com/cisagov/ansible-role-openvpn
   name: openvpn
+  version: feature/use-ufw-for-nat
```
Comment (mcdonnnj) [inline src/requirements.yml:23]: Wasn't this merged almost a month ago in https://github.com/cisagov/ansible-role-openvpn/pull/24?

**Review by mcdonnnj — APPROVED** [review]
Strong work 💪 Please see my one comment about the version of a requirement.

File: src/requirements.yml (hunk 2 above)
Comment (dav3r) [reply src/requirements.yml]: You are eagle-eyed as usual @mcdonnnj! I switched back to using the main branch of `ansible-role-openvpn` with https://github.com/cisagov/openvpn-packer/pull/27/commits/afcc2f1154a03cdaf0b6ad06b74feb15708a8c98.

File: src/ufw.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            proto: udp
+            rule: allow
+            to_port: bootps
+    - name: Configure routed/forwarded traffic
+      block:
+        # CIS hardening demands that the default policy for routed
+        # packets be deny, but this rule allows all routed packets as
+        # if the default policy were allow.  I'm following the letter
+        # of the law, not the intent.  Next thing you know I'll be
+        # chasing ambulances.  :(
```
Comment (felddy) [inline src/ufw.yml:86]: 😢

**Review by felddy — APPROVED** [review]
This looks good..
I need to go look a the remote roles to understand the hardening more.

File: src/harden.yml (hunk 1 above)
Comment (dav3r) [reply src/harden.yml]: I confirmed with @jsf9k via chat that these lines are fine to be removed, so I did so in https://github.com/cisagov/openvpn-packer/pull/27/commits/c382b7703f2a5a7d333191e4e36953d0d95d223a.

**Pull request merged**